distributed/tests/test_utils_test.py

```diff
@@ -731,15 +732,27 @@ def test_raises_with_cause():
             raise RuntimeError("exception") from ValueError("cause")
 
 
-def test_worker_fail_hard(capsys):
-    @gen_cluster(client=True, nthreads=[("127.0.0.1", 1)])
-    async def test_fail_hard(c, s, a):
-        with pytest.raises(Exception):
-            await a.gather_dep(
-                worker="abcd", to_gather=["x"], total_nbytes=0, stimulus_id="foo"
-            )
+@gen_cluster(nthreads=[("", 1)])
+async def test_fail_hard(s, a):
+    with captured_logger("distributed.worker") as logger:
+        # Asynchronously kick off handle_acquire_replicas on the worker,
+        # which will fail
+        s.stream_comms[a.address].send(
+            {
+                "op": "acquire-replicas",
+                "who_has": {"x": ["abcd"]},
+                "stimulus_id": "foo",
+            },
+        )
+        while a.status != Status.closed:
+            await asyncio.sleep(0.01)
+
+    assert "missing port number in address 'abcd'" in logger.getvalue()
 
-    with pytest.raises(Exception) as info:
-        test_fail_hard()
 
-    assert "abcd" in str(info.value)
+@gen_cluster(nthreads=[("", 1)])
+async def test_fail_hard_reraises(s, a):
+    with pytest.raises(AttributeError):
+        a.handle_stimulus(None)
+    while a.status != Status.closed:
+        await asyncio.sleep(0.01)
```

Comment:
These tests will need a second pass later on - they only test the sync half of the decorator